### PR TITLE
[3.2] Mono: Make Godot provide its own WASM m2n trampolines

### LINKED ISSUE #36858: GD.Seed C# Black Screen HTML5
**Godot version:**
Mono 3.2, 3.2.1 rc, 3.2.2 rc (I tested it with all of them as they were getting released to make sure it's not fixed) 

**OS/device including version:**
Linux (pop-os), HTML5 in Google Chrome

**Issue description:**
C#/F# GD.Seed(1UL);  (same for GD.Seed(1);) 
produce black screen of death in HTML build.
The black screen does not get produced when doing a Linux build. 
Only when it is in the browser.
Moreover, the issues does not occur if you call seed from GDScript.

**Steps to reproduce:**
1. Create scene
2. Add C# script to the scene
3. Inside ready call GD.Seed(1); or GD.Seed(1UL);
4. 
Test in Browser => Black Screen
Test Normally => Works fine
5. (Optional) Try same thing with GDScript => works fine both in browser and normally 
Project zip at the bottom after pics.

**Minimal reproduction project:**
image
image
[seed_bug.zip](https://github.com/godotengine/godot/files/4298993/seed_bug.zip)

### PR DESCRIPTION
**Pull request by neikeq**
Back-port of #44105 to the 3.2 branch.

This depends on a custom Mono patch from this commit: godotengine/godot-mono-builds@0e312939bd0dc4b807cc15dbe76a7b65456ab928

Also fixes #36858

### CODE REVIEW

**Comment by neikeq** [comment]
This PR against the 3.2 branch is tested and ready to be merged. Only pending review.

File: modules/mono/editor/bindings_generator.cpp [hunk 1]
```diff
@@ -95,7 +95,7 @@
 #define C_METHOD_MONOARRAY_TO(m_type) C_NS_MONOMARSHAL "::mono_array_to_" #m_type
 #define C_METHOD_MONOARRAY_FROM(m_type) C_NS_MONOMARSHAL "::" #m_type "_to_mono_array"
 
-#define BINDINGS_GENERATOR_VERSION UINT32_C(11)
+#define BINDINGS_GENERATOR_VERSION UINT32_C(13)
```
Comment (akien-mga) [inline modules/mono/editor/bindings_generator.cpp:98]: Is the bump from 11 to 13 expected?

File: modules/mono/mono_gd/gd_mono_utils.h [hunk 2]
```diff
@@ -151,6 +154,21 @@ struct ScopeThreadAttach {
 	MonoThread *mono_thread;
 };
 
+template <typename... P>
```
Comment (akien-mga) [inline modules/mono/mono_gd/gd_mono_utils.h:157]: Is this C++11/14 compatible in the end or does this form of parameter pack require C++17? https://en.cppreference.com/w/cpp/language/parameter_pack

Comment (neikeq) [reply modules/mono/mono_gd/gd_mono_utils.h:157]: I think it's C++11. I only had to look for an alternative for the `if constexpr`s when backporting this.

File: modules/mono/editor/bindings_generator.cpp (hunk 1 above)
Comment (neikeq) [reply modules/mono/editor/bindings_generator.cpp:98]: This was back-ported from master so probably version 12 was some other change that wasn't back-ported. It doesn't matter too much as this is used together with the API hash.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**